### PR TITLE
Prevent first time critical errors

### LINKED ISSUE #458: Prevent first time critical errors
As discussed in Slack, we allow users to enable Redis Object Cache without determining if Redis is reachable. This can lead to 500 errors after the drop-in is placed in situ.

We should not allow Redis Object Cache to be enabled via the UI if Redis cannot be reached and surface any errors. Now might also be a good time to improve the Overview tab.

CleanShot 2023-04-24 at 13 24 56@2x

- [x] Overview title removed because it's redundant
- [x] Status severity reduced to a warning because not being enabled isn't an error. Also made the copy more friendly for those not familiar with drop-ins.
- [x] Drop-in row removed because it's redundant if the plugin isn't enabled
- [x] New chunky icons

CleanShot 2023-04-24 at 13 23 39@2x

- [x] New notice indicating that Redis cannot be enabled
- [x] Enable button is disabled if the Filesystem or Redis checks fail

CleanShot 2023-04-24 at 13 30 25@2x

CleanShot 2023-04-24 at 13 32 15@2x

- [x] Status row and Drop-in rows merged. More appropriate copy for the success state.

### PR DESCRIPTION
**Pull request by A5hleyRich**
Resolves #458

### CODE REVIEW

**Comment by tillkruss** [comment]
Should we include the custom error message when connections fail in this PR as well?

**Comment by A5hleyRich** [comment]
@tillkruss are you referring to `db-error.php`?

**Comment by tillkruss** [comment]
Yes. Really what I mean is displaying a nice message when a connection error occurs.

**Comment by A5hleyRich** [comment]
@tillkruss this is what's shown when `WP_REDIS_GRACEFUL` is `false`. Links to: https://github.com/rhubarbgroup/redis-cache/blob/develop/INSTALL.md

If a `db-error.php` file exists, that is shown instead. I wasn't sure if we'd want our own file, but `db-error.php` is more well known so gone with that for now.

CleanShot 2023-05-02 at 13 45 54@2x

**Comment by A5hleyRich** [comment]
@tillkruss this is ready for review. Video of the changes:


https://user-images.githubusercontent.com/1422996/236030643-c318bf81-ac2c-4e80-a3fc-d18fea112777.mp4

**Comment by tillkruss** [comment]
Fucking amazing.

File: includes/class-plugin.php [hunk 1]
```diff
@@ -571,6 +579,66 @@ public function get_redis_status() {
...
+            'port' => 6379,
+            'database' => 0,
+            'timeout' => 1,
+            'read_timeout' => 1,
+            'retry_interval' => null,
+            'persistent' => false,
+        ];
+
+        $settings = [
+            'scheme',
```
Comment (tillkruss) [inline includes/class-plugin.php:600]: The only feedback I have is this function seems unnecessary on the `Plugin`, I'll move it to `Predis.`

**Review by tillkruss — APPROVED** [review]

**Pull request merged**